src/olympia/files/models.py

```diff
@@ -123,14 +164,17 @@ def from_upload(cls, upload, version, parsed_data=None):
             'is_mozilla_signed_extension', False
         )
         file_.is_signed = file_.is_mozilla_signed_extension
-        file_.filename = file_.generate_filename()
-
-        file_.hash = file_.generate_hash(upload_path)
+        file_.hash = upload.hash
         file_.original_hash = file_.hash
         file_.manifest_version = parsed_data.get(
             'manifest_version', DEFAULT_MANIFEST_VERSION
         )
-        file_.save()
+        log.info(f'New file: {file_!r} from {upload!r}')
+
+        # FIXME if FileUpload also did things correctly I wouldn't have to do this...
```

Comment:
out of interest, what's the fix here?  Or is it a django bug?  (is it a known bug?)

Reply:
Not a Django bug, I should reword this more clearly. What I'm complaining about is that `FileUpload.path` is not a `FileField` either. Because of this, I have to do the whole `open`+`DjangoFile()` dance to get a `DjangoFile` instance.

Reply:
4e8fb2d1958d6f97cf8b8da7bdf7ba889296d7be